crates/autopilot/src/run_loop.rs

```diff
@@ -145,6 +145,42 @@ impl RunLoop {
                 }
             };
 
+            {
+                // Also calculate the score from the solution itself
+                // For now, don't actually use it for ranking, instead log differences to solver
+                // provided score for debugging purposes.
+                let auction_prices = auction
+                    .prices
+                    .iter()
+                    .map(|(k, v)| {
+                        (
+                            (*k).into(),
+                            auction::Price::new(domain::eth::Ether(*v)).unwrap(),
+                        )
+                    })
+                    .collect();
+                let policies = auction
+                    .orders
+                    .iter()
+                    .map(|order| (order.uid, order.protocol_fees.clone()))
+                    .collect();
+                let settlement =
+                    domain::settlement::Settlement::from_solution(solution, &auction, auction_id)
+                        .map(|s| s.score(&auction_prices, &policies));
```

Comment:
We already pass in the entire auction to create the `Settlement`. If the `Settlement` is an immutable object anyway why don't we already compute the score in the constructor?

Reply:
Because `Settlement` has another constructor where it is constructed solely from `calldata`.